src/js/utils/defineCustomElement.js

```diff
@@ -1,6 +1,11 @@
+import { isServer, Window } from './browser-env.js';
+
 export function defineCustomElement(name, element) {
-  if (!window.customElements.get(name)) {
-    window.customElements.define(name, element);
-    window[element.name] = element;
+  if (isServer()) {
+    return;
+  }
+  if (!Window.customElements.get(name)) {
```

Comment:
We don't really need the Window part here if we've already returned if isServer. Unless you're going for consistency of window usage.

Reply:
>  you're going for consistency of window usage.

The idea was to go for consistency using `Window` everywhere